src/Razor/src/Microsoft.AspNetCore.Razor.LanguageServer/Semantic/Services/DefaultRazorSemanticTokensInfoService.cs

```diff
@@ -26,19 +27,14 @@ namespace Microsoft.AspNetCore.Razor.LanguageServer.Semantic
 {
     internal class DefaultRazorSemanticTokensInfoService : RazorSemanticTokensInfoService
     {
-        // These caches are created to restrict memory growth.
-        // We need to keep track of the last couple of requests for use in previousResultId,
-        // but if we let them grow unbounded it could quickly allocate a lot of memory.
-        // Solution: in-memory caches
-        private readonly MemoryCache<string, (VersionStamp SemanticVersion, IReadOnlyList<int> Data)> _semanticTokensCache = new(); // For Razor docs
-        private readonly MemoryCache<string, IReadOnlyList<int>> _csharpGeneratedSemanticTokensCache = new(); // For C# generated docs
-
         private readonly ClientNotifierServiceBase _languageServer;
+        private readonly RazorDocumentMappingService _documentMappingService;
         private readonly ProjectSnapshotManagerDispatcher _projectSnapshotManagerDispatcher;
         private readonly DocumentResolver _documentResolver;
         private readonly DocumentVersionCache _documentVersionCache;
         private readonly ILogger _logger;
-        private readonly RazorDocumentMappingService _documentMappingService;
+        private readonly SemanticTokensCache<VersionedSemanticTokens> _razorDocTokensCache;
+        private readonly SemanticTokensCache<SemanticTokens> _csharpGeneratedDocTokensCache;
```

Comment:
Would it be bad if `SemanticTokens` was `VersionedSemanticTokens`? Wondering if it could simplify the underlying data structure so it didn't have to be generic